### DISCUSSION TITLE
What is the correct way of checking if a type is already registered?

### DISCUSSION THREAD

**Discussion by JvJ** [pybind/pybind11#3961] [Q&A]
I'm working on a project that involves generating C++ code, and there are many different aliases that end up resolving to the same type.  This is causing "type already registered" errors.

I'm unsure of what is the correct way to determine whether a type is already registered.

I am currently attempting to use `py::detail::get_type_info` to get a `py::detail::type_info` structure.  However, the fact that this structure is defined in a header called "internals" makes me think it's not a good idea to use it.

Is there another, better way?

**Comment by alemuntoni** [comment]
I'm facing the same issue. 
Is there any way to check if a type is already registered in a module?
